File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/ParkOnLine.java
package org.firstinspires.ftc.teamcode;

import com.qualcomm.robotcore.eventloop.opmode.Autonomous;
import com.qualcomm.robotcore.eventloop.opmode.LinearOpMode;

@SuppressWarnings("unused")
@Autonomous(name = "Park on line")
public class ParkOnLine extends LinearOpMode {
    private static final Double INCHES_PER_SECOND = 52.5;
    RobotHardware robot;

    @Override
    public void runOpMode() {
        robot = new RobotHardware();
        robot.init(hardwareMap, "");
        waitForStart();
        robot.driveFor(90.0);
    }

}

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/ParkOnLine.kt
package org.firstinspires.ftc.teamcode

import com.qualcomm.robotcore.eventloop.opmode.Autonomous
import com.qualcomm.robotcore.eventloop.opmode.LinearOpMode

@Autonomous(name = "Park on line")
class ParkOnLine : LinearOpMode() {
    override fun runOpMode() {
        val robot = RobotHardware()
        robot.init(hardwareMap, "")
        waitForStart()
        robot.chassis(doubleArrayOf(-1.0, 0.0, 0.0))
        sleepInches(35.0)
        robot.brake()
        sleep(0)
        robot.startFlywheels()
        robot.startIntake()
        sleep(500)
        robot.fire()
        sleep(500)
        robot.fire()
        sleep(500)
        robot.fire()
        sleep(500)
        robot.stopFlywheels()
        robot.stopIntake()
        robot.chassis(doubleArrayOf(-1.0, 0.0, 0.0))
        sleepInches(10.0)
        robot.brake()
    }

    private fun sleepInches(inches: Double) {
        sleep((inches / (RobotHardware.INCHES_PER_SECOND / 1000)).toLong())
    }

}